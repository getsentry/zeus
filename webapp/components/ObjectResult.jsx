import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {css} from '@emotion/core';
import styled from '@emotion/styled';

import Icon from './Icon';

import {
  MdCheckCircle,
  MdAvTimer,
  MdHistory,
  MdCancel,
  MdError,
  MdRemoveCircle,
  MdHelp
} from 'react-icons/md';

export default class ObjectResult extends Component {
  static propTypes = {
    data: PropTypes.shape({
      result: PropTypes.string.isRequired,
      status: PropTypes.string
    }),
    size: PropTypes.number
  };

  static defaultProps = {
    size: 16
  };

  getIcon(result, status) {
    let {size} = this.props;
    switch (status) {
      case 'queued':
        return <MdHistory size={size} />;
      case 'in_progress':
        return <MdAvTimer size={size} />;
      default:
        if (status === 'finished' && result === 'unknown') return <MdHelp size={size} />;
        switch (result) {
          case 'passed':
            return <MdCheckCircle size={size} />;
          case 'aborted':
            return <MdCancel size={size} />;
          case 'failed':
            return <MdRemoveCircle size={size} />;
          case 'errored':
            return <MdError size={size} />;
          default:
            return null;
        }
    }
  }

  render() {
    if (!this.props.data) {
      return null;
    }
    let {result, status} = this.props.data;
    return (
      <ResultIcon status={status} result={result} mr>
        {this.getIcon(result, status)}
      </ResultIcon>
    );
  }
}

export const ResultIcon = styled(Icon)`
  ${props => {
    switch (props.result) {
      case 'passed':
        return css`
          svg {
            color: #76d392;
          }
        `;
      case 'errored':
        return css`
          color: #f0a05b;
          svg {
            color: #f0a05b;
          }
        `;
      case 'failed':
        return css`
          color: #f05b5b;
          svg {
            color: #f05b5b;
          }
        `;
      default:
        return css`
          svg {
            color: #bfbfcb;
          }
        `;
    }
  }};
`;
