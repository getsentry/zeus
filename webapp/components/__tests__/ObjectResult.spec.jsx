import React from 'react';
import {render} from 'enzyme';

import ObjectResult from '../ObjectResult';

describe('ObjectResult', () => {
  it('renders queued', () => {
    const tree = render(
      <ObjectResult
        data={{
          status: 'queued',
          result: 'unknown'
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders in-progress', () => {
    const tree = render(
      <ObjectResult
        data={{
          status: 'in_progress',
          result: 'unknown'
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders finished unknown', () => {
    const tree = render(
      <ObjectResult
        data={{
          status: 'finished',
          result: 'unknown'
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders finished passed', () => {
    const tree = render(
      <ObjectResult
        data={{
          status: 'finished',
          result: 'passed'
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders finished failed', () => {
    const tree = render(
      <ObjectResult
        data={{
          status: 'finished',
          result: 'failed'
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders finished errored', () => {
    const tree = render(
      <ObjectResult
        data={{
          status: 'finished',
          result: 'errored'
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
