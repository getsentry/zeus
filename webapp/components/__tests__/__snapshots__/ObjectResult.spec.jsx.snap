// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectResult renders finished errored 1`] = `
<div
  class="sc-bdVaJa bGveho"
  result="errored"
>
  <svg
    fill="currentColor"
    height="16"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="16"
  >
    <g>
      <path
        d="m21.6 21.6v-10h-3.2v10h3.2z m0 6.8v-3.4h-3.2v3.4h3.2z m-1.6-25c9.2 0 16.6 7.4 16.6 16.6s-7.4 16.6-16.6 16.6-16.6-7.4-16.6-16.6 7.4-16.6 16.6-16.6z"
      >
        
      </path>
    </g>
  </svg>
</div>
`;

exports[`ObjectResult renders finished failed 1`] = `
<div
  class="sc-bdVaJa imLzDX"
  result="failed"
>
  <svg
    fill="currentColor"
    height="16"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="16"
  >
    <g>
      <path
        d="m28.4 21.6v-3.2h-16.8v3.2h16.8z m-8.4-18.2c9.2 0 16.6 7.4 16.6 16.6s-7.4 16.6-16.6 16.6-16.6-7.4-16.6-16.6 7.4-16.6 16.6-16.6z"
      >
        
      </path>
    </g>
  </svg>
</div>
`;

exports[`ObjectResult renders finished passed 1`] = `
<div
  class="sc-bdVaJa ePPuOg"
  result="passed"
>
  <svg
    fill="currentColor"
    height="16"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="16"
  >
    <g>
      <path
        d="m16.6 28.4l15-15-2.3-2.5-12.7 12.7-5.9-5.9-2.3 2.3z m3.4-25c9.2 0 16.6 7.4 16.6 16.6s-7.4 16.6-16.6 16.6-16.6-7.4-16.6-16.6 7.4-16.6 16.6-16.6z"
      >
        
      </path>
    </g>
  </svg>
</div>
`;

exports[`ObjectResult renders finished unknown 1`] = `
<div
  class="sc-bdVaJa kKlySH"
  result="unknown"
>
  <svg
    fill="currentColor"
    height="16"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="16"
  >
    <g>
      <path
        d="m25.1 18.8c0.9-1 1.5-2.3 1.5-3.8 0-3.7-2.9-6.6-6.6-6.6s-6.6 2.9-6.6 6.6h3.2c0-1.8 1.6-3.4 3.4-3.4s3.4 1.6 3.4 3.4c0 0.9-0.4 1.7-1.1 2.3l-2 2.2c-1.2 1.2-1.9 2.8-1.9 4.6v0.9h3.2c0-2.5 0.8-3.4 2-4.7z m-3.5 12.8v-3.2h-3.2v3.2h3.2z m-1.6-28.2c9.2 0 16.6 7.4 16.6 16.6s-7.4 16.6-16.6 16.6-16.6-7.4-16.6-16.6 7.4-16.6 16.6-16.6z"
      >
        
      </path>
    </g>
  </svg>
</div>
`;

exports[`ObjectResult renders in-progress 1`] = `
<div
  class="sc-bdVaJa kKlySH"
  result="unknown"
>
  <svg
    fill="currentColor"
    height="16"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="16"
  >
    <g>
      <path
        d="m10 20c0-0.9 0.7-1.6 1.6-1.6s1.8 0.7 1.8 1.6-0.8 1.6-1.8 1.6-1.6-0.7-1.6-1.6z m20 0c0 0.9-0.7 1.6-1.6 1.6s-1.8-0.7-1.8-1.6 0.8-1.6 1.8-1.6 1.6 0.7 1.6 1.6z m-11.6-15h1.6c8.3 0 15 6.7 15 15s-6.7 15-15 15-15-6.7-15-15c0-4.9 2.3-9.2 6-12v0l11.3 11.3-2.3 2.3-9.1-8.9c-1.6 2-2.5 4.5-2.5 7.3 0 6.5 5.1 11.6 11.6 11.6s11.6-5.1 11.6-11.6c0-5.9-4.3-10.8-10-11.6v3.2h-3.2v-6.6z m0 23.4c0-1 0.7-1.8 1.6-1.8s1.6 0.8 1.6 1.8-0.7 1.6-1.6 1.6-1.6-0.7-1.6-1.6z"
      >
        
      </path>
    </g>
  </svg>
</div>
`;

exports[`ObjectResult renders queued 1`] = `
<div
  class="sc-bdVaJa kKlySH"
  result="unknown"
>
  <svg
    fill="currentColor"
    height="16"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="16"
  >
    <g>
      <path
        d="m20 13.4h2.5v7l5.9 3.5-1.3 2-7.1-4.3v-8.2z m1.6-8.4q6.2 0 10.6 4.4t4.4 10.6-4.4 10.6-10.6 4.4-10.5-4.4l2.3-2.4q3.5 3.4 8.2 3.4 4.9 0 8.3-3.4t3.5-8.2-3.5-8.2-8.3-3.4-8.2 3.4-3.4 8.2h5l-6.7 6.7-0.2-0.2-6.5-6.5h5q0-6.2 4.5-10.6t10.5-4.4z"
      >
        
      </path>
    </g>
  </svg>
</div>
`;
